docs/js-dev-standards.md

```diff
@@ -0,0 +1,101 @@
+* __Modern__ We lean into the latest features of the JavaScript ecosystem, including TypeScript, ESM, and modern language features.
+
+## Project Structure
+Our projects are structured as monorepos using Yarn 3+ (or whatever is latest). Our software is usually composed of many moving parts: libraries, demo applications, full-blown applications, test suites, and more. To manage these in a single repository, we use Yarn workspaces.
+
+### JavaScript-Only Monorepos
+For single-language monorepos, workspaces are created under the `packages` folder. For example, a monorepo with a library and a demo application would look like this:
+
+```text
+- packages
+	- my-library
+		- package.json
+	- my-demo
+		- package.json
+```
+
+### Multi-Language Monorepos
+When authoring monorepos with multiple languages, Yarn workspaces are created under the `javascript` folder, and other languages would follow a similar pattern. For example, a monorepo with a library, demo application, and a Go backend would look like this:
+
+```text
+- javascript
+  - my-library
+		- package.json
+	- my-demo
+		- package.json
+- go
+	- my-backend
+		- go.mod
+		- go.sum
+		- main.go
+```
+
+The rest of this document will focus primarily on JavaScript.
+
+## Common Build Tasks
+The following top-level tasks expected to exist in all JavaScript projects. 
+
+* __clean__ - this task cleans the project, removing any artifacts or temporary files.
+* __release__ - this task applies semver changes, executes build verification, and publishes artifacts to the package registry.
+* __ci__ - this task is used to chain together an end-to-end continuous verification pipeline. It should run all tests, linting, and other verification tasks.
+* __start__ - this task starts any webapps or servers in the project in development mode.
+* __build__ - this task builds artifacts and performs typechecking for library packages within the project.
+* __test__ - this task runs all tests in the project.
+* __check__ - this task checks code for linting and formatting issues.
+* __fix__ - this task checks code for linting and formatting issues and attempts to fix as many as possible.
+* __update_sdks__ this is a utility command that's occasionally used to update yarn sdks (e.g. `yarn dlx @yarnpkg/sdks vscode`)
+
+## CI Tasks
+* Code Verification (e.g. `yarn ci`) - perform basic code verification across the project. This task should also verify that the git repository is clean when it is finished.
+* SemVer Verification (e.g. `yarn version check`)
+* website/docsite release processes
+
+
+## Tooling
+The following tools are used to manage monorepos.
+
+### Yarn
+As mentioned above, we use Yarn workspaces to manage monorepos. Yarn is a package manager for JavaScript, and is used to install dependencies, run scripts, and more. Yarn is also used to manage the monorepo, and is the primary tool used to run scripts and commands.
+
+#### Key, idiomatic, Yarn features
```

Comment:
We are moving away from using cached deps, right?

Reply:
They are not committed into the repository anymore (e.g. Yarn Zeroinstall) , but we still use PnP